### PR TITLE
Some OpenSSL issues

### PR DESCRIPTION
**Pull request by tiennou**
Reading #4644 was eerily familiar, so it made me start looking for "that patch I had written but now I can't find it" 😉.

This is extracted from #4786, I'm not sure if it is the complete story, but at least it would help downstream (and us) to pinpoint the reason for those failures.

### CODE REVIEW

File: src/streams/openssl.c [hunk 1]
```diff
@@ -373,10 +373,10 @@ static int ssl_set_error(SSL *ssl, int error)
 	switch (err) {
 	case SSL_ERROR_WANT_CONNECT:
 	case SSL_ERROR_WANT_ACCEPT:
-		giterr_set(GITERR_NET, "SSL error: connection failure");
+		giterr_set(GITERR_SSL, "SSL error: connection failure");
```
Comment (pks-t) [inline src/streams/openssl.c]: I think these changes make sense. The other secure streams also use `GITERR_SSL` for various SSL-related failures

File: src/streams/openssl.c [hunk 2]
```diff
@@ -602,6 +600,8 @@ int openssl_connect(git_stream *stream)
 	if ((ret = SSL_connect(st->ssl)) <= 0)
 		return ssl_set_error(st->ssl, ret);
 
+	st->connected = true;
```
Comment (pks-t) [inline src/streams/openssl.c]: Makes sense, too. One might wonder about whether the BIO will still get free'd if we never call `SSL_shutdown`. But documentation of `SSL_set_bio` states that BIO's ownership is transferred to the SSL handle and will get free'd with `BIO_free_all` as soon the SSL handle is free'd.

**Comment by ethomson** [comment]
Thanks!

**Comment by tiennou** [comment]
@ethomson Though it's minor, also next-release (unless it was backported).

```
* Fix some issues with the error-reporting in the OpenSSL backend
```

**Pull request merged**